File: app/src/main/java/com/neome/java/api/ent/base/dto/DtoEntActionRowInsert.java
// neome.ai API - do not change
//

package com.neome.java.api.ent.base.dto;

import com.google.gson.JsonElement;
import com.neome.java.api.meta.base.Types.MetaIdComp;
import com.neome.java.api.meta.base.Types.MetaIdForm;
import com.neome.java.api.meta.base.Types.MetaIdLayoutForm;
import com.neome.java.api.meta.base.Types.MetaIdSpreadsheet;

import org.jetbrains.annotations.Nullable;

import java.util.Map;

@SuppressWarnings("unused")
public class DtoEntActionRowInsert extends DtoEntAction
{
  @Nullable
  public Map<MetaIdComp, JsonElement> defaultValueMap;

  @Nullable
  public MetaIdLayoutForm formEditorLayoutId;

  @Nullable
  public Boolean hasPartitions;

  @Nullable
  public MetaIdLayoutForm mobileFormEditorLayoutId;

  @Nullable
  public Boolean sendMessageToInbox;

  public MetaIdForm spreadsheetFormId;

  public MetaIdSpreadsheet spreadsheetId;
}

File: app/src/main/java/com/neome/api/ent/base/dto/DtoEntActionRowInsert.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.ent.base.dto

import kotlinx.serialization.json.JsonElement
import com.neome.api.ent.base.dto.DtoEntAction
import com.neome.api.meta.base.Types.EnumDefnKindFormRenderingMode
import com.neome.api.meta.base.Types.MetaIdComp
import com.neome.api.meta.base.Types.MetaIdForm
import com.neome.api.meta.base.Types.MetaIdLayoutForm
import com.neome.api.meta.base.Types.MetaIdSpreadsheet

interface DtoEntActionRowInsert : DtoEntAction
{
  val defaultValueMap: Map<MetaIdComp, JsonElement>?
  val formEditorLayoutId: MetaIdLayoutForm?
  val formRenderingModeKind: EnumDefnKindFormRenderingMode?
  val hasPartitions: Boolean?
  val mobileFormEditorLayoutId: MetaIdLayoutForm?
  val sendMessageToInbox: Boolean?
  val spreadsheetFormId: MetaIdForm
  val spreadsheetId: MetaIdSpreadsheet
}